### PR TITLE
Fix ContextMenus

### PR DESCRIPTION
**Pull request by Plerx2493**
# Summary
Fix context menu parsing in DiscordClient.Dispatch and fix filtering in SlashCommandsExtension#InteractionHandler

# Notes 
Tested

### CODE REVIEW

**Comment by Plerx2493** [comment]
Waiting on additional testing from a reporting user

**Comment by Plerx2493** [comment]
Testing was succesful, waiting on a review from another dev

**Review by OoLunar — APPROVED** [review]

**Pull request merged**
